dev-util/bats/bats-1.9.0.ebuild

```diff
@@ -0,0 +1,44 @@
+# Copyright 1999-2023 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+inherit multiprocessing optfeature
+
+MY_PN="bats-core"
+DESCRIPTION="Bats-core: Bash Automated Testing System"
+HOMEPAGE="https://github.com/bats-core/bats-core/"
+SRC_URI="https://github.com/${MY_PN}/${MY_PN}/archive/v${PV}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="MIT"
+SLOT="0"
+KEYWORDS="~amd64 ~arm ~arm64 ~hppa ~ppc ~ppc64 ~riscv ~sparc ~x86"
+
+DEPEND="app-shells/bash:*"
+RDEPEND="${DEPEND}"
+
+S="${WORKDIR}/${MY_PN}-${PV}"
+
+src_test() {
+	# magic number to pass "parallelity factor is met exactly"
+	local my_jobs=5
```

Comment:
@martin-schulze-vireso 

Gentoo Linux does not just ship this package but also has "upstream testing" support. We used to have to make a difference on that testing depending on if "parallel" is installed or not. Somewhere between 1.8.2 and 1.9.0 we have to inherit the "magic" "5" to pass the test "parallelity factor is met exactly"

Not the biggest issue, but we have to hardcode a magic number and can not test "-j 64". Please let me know if i do this all the wrong way. Best for "downstream testing" would be "upstream testing" transparently taking care of "is parallel usable?" and taking care of "magic numbers" that some tests might need.

Reply:
sorry for that direct "mention" but upstream issue templates seemed a little restrictive and i did not understand how to place that

Reply:
Please open a bug ticket for that and reference this discussion, I'll process from there.

Reply:
was a mistake on my side, the test that failed is time-based, we have nice levels and i was heavily overcommited because i had distcc MAKEOPTS ... j32 on a machine that should rather have j4

Reply:
let us hope that nobody else runs into this timeout, i would wait for potential gentoo bug reports and now just call the tests as we used to

Reply:
Okay, sounds plausible. I was wondering why the test with explizit parallelity factor was affected.

Reply:
i actually switched from "makeopts_jobs" to "get_nproc" now to even deal with this

unfortunately i could not repro outside the context of gentoo portage, so i have not much to report to upstream ... except for timing based tests being a strong assumption that could fail in some weird cases